OPHD/StructureManager.h

```diff
@@ -37,6 +37,38 @@ class SKey
 };
 
 
+/**
+ * Object suitable for range-based for loops.
+ * This class allows code to iterate over instances of a given component
+ * without exposing the internal component storage container type.
+ */
+template<typename ComponentTy>
+class ComponentRange
```

Comment:
There are now 3 top level classes in one file. Might be good to either break them out into separate files, or make them inner classes.